### ISSUE TITLE
Provide a self-contained plugin file

### ISSUE THREAD

**Issue by drzraf** [kjellm/munin-mysql#92]
This great plugin would be more used if it could fit in a self-contained perl-script rather than a whole make/installation.
Hint: The make could make concatenation of the perl-code of the libraries?

**Comment by szepeviktor** [comment]
Or it could be a proper Debian package.

**Comment by roidelapluie** [comment]
I'd rather have proper packages

**Comment by meersjo** [comment]
I agree, proper packages would be superior, especially since they also can take into account the differing directory structures on various distributions.

Also, it's been a long time since I did some proper perl, but I suspect that perl modules depend on being stored in separate files?

**Comment by szepeviktor** [comment]
The first step could be a module on CPAN
https://perldoc.perl.org/perlnewmod.html

**Comment by meersjo** [comment]
I don't believe that is appropriate - CPAN modules are intended for generic functionality; this is a specific software that is only useful in it's own context - monitoring a MySQL installation. 

Now, if you were to generalize this into perl::mMnin::monitorThing that would be OK, but given that we already import perl::Munin, which is basically that, I don't think that would be particularly worthwile. 

> From: "Viktor Szépe" <notifications@github.com>
> To: "kjellm/munin-mysql" <munin-mysql@noreply.github.com>
> Cc: "meersjo" <vegivamp+github@tuxera.be>, "Comment"
> <comment@noreply.github.com>
> Sent: Friday, 7 April, 2017 13:00:48
> Subject: Re: [kjellm/munin-mysql] Provide a self-contained plugin file (#92)

> The first step could be a module on CPAN
> https://perldoc.perl.org/perlnewmod.html

> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub , or mute the thread .

-- 
Unhappiness is discouraged and will be corrected with kitten pictures.

**Comment by szepeviktor** [comment]
OK.
Executable files could be packaged also.
example: https://anonscm.debian.org/cgit/pkg-perl/packages/libdbi-perl.git/tree/Makefile.PL#n91
another one: https://metacpan.org/source/TIMB/DBI-1.636/Makefile.PL#L91
docs: https://perldoc.perl.org/ExtUtils/MakeMaker.html#Using-Attributes-and-Parameters See EXE_FILES

Actually I don't speak PERL.

**Comment by szepeviktor** [comment]
I think packaging is mainly for maintenance (no stray files) and updates.

Installation could be `cpan Munin::MySQL::Graph` or `apt-get install munin-plugin-mysql` using [dh-make-perl](https://packages.debian.org/stretch/dh-make-perl)

**Comment by kjellm** [comment]
I don't think you can get cpan to install stuff to `/etc/munin`. We need to put `mysql.conf` under `/etc/munin/plugin-conf.d/`. And `munin` under `/etc/munin/plugins`. So we would need an extra install step even if we make a CPAN module.

**Comment by szepeviktor** [comment]
> So we would need an extra install step even if we make a CPAN module.

Currently `mysql.conf` need manual configuration also.

When configuring `mysql.conf` one may create a symlink for `mysql` in `/etc/munin/plugins` (as in Debian)

**Comment by szepeviktor** [comment]
### Installation

1. Please install the perl module: `cpan Munin::MySQL::Graph`
1. Enter your DB credentials in `/etc/munin/plugin-conf.d/mysql.conf`. Skeleton configuration file can be found in ...
1. Activate the plugin by creating a symlink: `ln -s .../mysql /etc/munin/plugins/`
